core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -333,8 +351,11 @@ private[spark] class TaskSchedulerImpl(
       availableCpus: Array[Int],
       availableResources: Array[Map[String, Buffer[String]]],
       tasks: IndexedSeq[ArrayBuffer[TaskDescription]],
-      addressesWithDescs: ArrayBuffer[(String, TaskDescription)]) : Boolean = {
+      addressesWithDescs: ArrayBuffer[(String, TaskDescription)])
+    : (Boolean, Boolean, Option[TaskLocality]) = {
```

Comment:
shall we add some doc for this method to explain the return type?

Reply:
👍